SecretSanta/test/SecretSanta.Api.Tests/Controllers/UserControllerTests.cs

```diff
@@ -5,16 +5,317 @@
 using System.Net.Http;
 using System.Net.Http.Json;
 using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.Testing;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SecretSanta.Api.Dto;
+using SecretSanta.Api.Tests.Business;
+using SecretSanta.Data;
 
 namespace SecretSanta.Api.Tests.Controllers
 {
     [TestClass]
     public class UserControllerTests
     {
-        
+        private static CustomWebApplicationFactory Factory { get; } = new();
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void UsersController_WithNullParameter_ThrowsException()
+        {
+            //Arrange - Nothing to arrange
+
+            //Act
+            Api.Controllers.UsersController controllerTemp = new(null!);
+
+            //Assert - Handled in tags
+        }
+
+        [TestMethod]
+        public void UsersController_WithValidParameter_DoesNotThrowException()
+        {
+            //Arrange - Nothing to arrange
+
+            //Act
+            Api.Controllers.UsersController controllerTemp = new(new TestableUserRepository());
+
+            //Assert - Nothing to assert
+        }
+
+        [TestMethod]
+        public async Task Get_WithData_ReturnsAccurateUserList()
+        {
+            //Arrange
+            TestableUserRepository testableRepo = Factory.UserRepo;
+            HttpClient client = Factory.CreateClient();
+
+            User user0 = new User() { Id = 998, FirstName = "Who", LastName = "Knows" };
+            User user1 = new User() { Id = 999, FirstName = "I", LastName = "Do" };
+
+            testableRepo.UserList = new List<User>() { user0, user1, };
+
+            //Act
+            HttpResponseMessage response = await client.GetAsync("/api/users");
+            List<User>? userList = await response.Content.ReadFromJsonAsync<List<User>?>();
+            
+            //Assert
+            response.EnsureSuccessStatusCode();
+
+            if(userList is null)
+            {
+                Assert.Fail("No List was received. \'" + nameof(userList) + "\' is null.");
+                return;
+            }
```

Comment:
Typically you want to avoid any conditional statements in your tests like this. A better way to handle this case would be to simply assert that the `userList` is not null.
```suggestion
            Assert.IsNotNull(userList);
```